cmake/platformChecks.cmake

```diff
@@ -16,6 +17,10 @@ CHECK_TYPE_SIZE("max_align_t" MAX_ALIGN_T)
 
 set(CMAKE_EXTRA_INCLUDE_FILES "cstddef")
 CHECK_TYPE_SIZE("std::max_align_t" STD_MAX_ALIGN_T LANGUAGE CXX)
+set(CMAKE_EXTRA_INCLUDE_FILES "type_traits")
+set(CMAKE_REQUIRED_FLAGS "-std=c++11") # required for g++ <= 5
```

Comment:
Since this is only relevant for GCC it should be surrounded by a check for that compiler. `if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")` should work since that is the same string our toolchain detection uses.

Since this applies to all the C++ checks, could you insert that code above the other `CHECK_TYPE_SIZE`? That will make the results more consistent.